docs/topics/dokkatoo-migration.md

```diff
@@ -0,0 +1,376 @@
+
+## Migrate your project
+
+After updating the Dokka Gradle plugin to v2, follow the migration steps applicable to your project.
+
+### Adjust configuration options
+
+DGP v2 introduces some changes in the Gradle configuration options. In the `build.gradle.kts` file, adjust the configuration 
+options according to your project setup: 
+
+* **New top-level DSL configuration:** Replace the old configuration syntax with the 
+  new top-level `dokka {}` DSL configuration. For example:
+
+  Previous configuration:
+
+  ```kotlin
+  tasks.withType<DokkaTask>().configureEach {
+      dokkaSourceSets {
+          named("main") {
+              moduleName.set("Project Name")
+              includes.from("README.md")
+              sourceLink {
+                  localDirectory.set(file("src/main/kotlin"))
+                  remoteUrl.set(URL("https://example.com/src"))
+                  remoteLineSuffix.set("#L")
+              }
+          }
+      }
+  }
+
+  tasks.dokkaHtml {
+      pluginConfiguration<DokkaBase, DokkaBaseConfiguration> {
+          customStyleSheets.set(listOf("styles.css"))
+          customAssets.set(listOf("logo.png"))
+          footerMessage.set("(c) Your Company")
+      }
+  }
+  ```
+ 
+  New configuration:
+
+  ```kotlin
+  dokka {
+      moduleName = "Project Name"
+      dokkaSourceSets.main {
+          includes.from("README.md")
+          sourceLink {
+              localDirectory = file("src/main/kotlin")
+              remoteUrl = URL("https://example.com/src")
+              remoteLineSuffix = "#L"
+          }
+      }
+      pluginsConfiguration.html {
+          customStyleSheets.from("styles.css")
+          customAssets.from("logo.png")
+          footerMessage = "(c) Your Company"
+      }
+  }
+  ```
+
+* **Visibility settings:**  The `documentedVisibilities` property has changed from `Visibility.PUBLIC` to `VisibilityModifier.Public`.
+
+  Previous configuration:
+
+  ```kotlin
+  import org.jetbrains.dokka.DokkaConfiguration.Visibility
+  ...
+  documentedVisibilities.set(
+      setOf(Visibility.PUBLIC)
+  )  
+  ```
+
+  New configuration:
+
+  ```kotlin
+  import org.jetbrains.dokka.gradle.engine.parameters.VisibilityModifier
+  ...
+  documentedVisibilities.set(
+      setOf(VisibilityModifier.Public)
+  )
+  ```
+
+* **External documentation link:** The source link configuration has [changed](https://docs.gradle.org/current/userguide/upgrading_version_8.html#deprecated_invalid_url_decoding). 
+    The remote URL is now specified using `URI` instead of `URL`.
+
+  Previous configuration:
+  
+    ```kotlin
+    remoteUrl.set(URL("https://github.com/your-repo"))   
+    ```
+  
+  New configuration:
+  
+    ```kotlin
+    remoteUrl.set(URI("https://github.com/your-repo"))  
+    ```
+
+* **Custom assets:** The [`customAssets`](dokka-html.md#customize-assets) property now uses collections of files (`FileCollection`) instead of lists (`var List<File>`).
```

Comment:
Suggestion: What about adding a link to the Gradle docs about FileCollections?

```suggestion
* **Custom assets:** The [`customAssets`](dokka-html.md#customize-assets) property now uses collections of files ([`FileCollection`](https://docs.gradle.org/8.10/userguide/lazy_configuration.html#working_with_files_in_lazy_properties)) instead of lists (`var List<File>`).
```